src/NHibernate/Linq/LinqExtensionMethods.cs

```diff
@@ -15,95 +15,6 @@ namespace NHibernate.Linq
 {
 	public static class LinqExtensionMethods
 	{
-		private static readonly Dictionary<string, MethodInfo> cachableQueryableMethods;
```

Comment:
All putted in `ReflectionCache`. Granted, only part of them are used elsewhere, but that avoids mixing here "in-place" resolution and resolution in cache.